Loop/Managers/Alerts/AlertStore.swift

```diff
@@ -0,0 +1,149 @@
+//
+//  PersistentDeviceAlertLog.swift
+//  Loop
+//
+//  Created by Rick Pasetto on 5/11/20.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import CoreData
+import LoopKit
+
+public class AlertStore {
+    
+    public enum AlertStoreError: Error {
+        case notFound
+    }
+       
+    private let managedObjectContext: NSManagedObjectContext
+
+    private let persistentContainer: NSPersistentContainer
+        
+    private let log = DiagnosticLog(category: "AlertStore")
+
+    private let dataAccessQueue = DispatchQueue(label: "com.loop.AlertStore.dataAccessQueue", qos: .utility)
+    
+    public init(storageFileURL: URL? = nil) {
+        managedObjectContext = NSManagedObjectContext(concurrencyType: .privateQueueConcurrencyType)
+        managedObjectContext.mergePolicy = NSMergeByPropertyStoreTrumpMergePolicy
+        managedObjectContext.automaticallyMergesChangesFromParent = true
+
+        let storeDescription = NSPersistentStoreDescription()
+        if let storageFileURL = storageFileURL {
+            storeDescription.url = storageFileURL
+            print("AlertStore: \(storageFileURL)")
+        } else {
+            storeDescription.type = NSInMemoryStoreType
+        }
+        persistentContainer = NSPersistentContainer(name: "AlertStore")
+        persistentContainer.persistentStoreDescriptions = [storeDescription]
+        persistentContainer.loadPersistentStores { description, error in
+            if let error = error {
+                fatalError("Unable to load persistent stores: \(error)")
+            }
+        }
+        managedObjectContext.persistentStoreCoordinator = persistentContainer.persistentStoreCoordinator
+    }
+}
+
+// MARK: Alert Recording
+
+extension AlertStore {
+    
+    public func recordIssued(alert: DeviceAlert, at timestamp: Date = Date(), completion: ((Result<Void, Error>) -> Void)? = nil) {
```

Comment:
[nit] I'm getting some naming confusion here. This is the `AlertStore` that deals with `StoredAlerts` with a parameter name `alert`, but a type of `DeviceAlert`?

Reply:
yeah...all of this used to be called `DeviceAlert*`, but I see that now my renaming is making things a little more confusing. I don't know where or when we'll have "non-device alerts" as Pete suggested earlier in this PR, but I'd imagine we'd add functions here to record those as well.  It may involve a refactor where there might be a generic `Alert` (protocol?), of which `DeviceAlert` is one.  At any rate, until we start building those "non-device alerts" I will leave this as-is.

Reply:
What about alerts that just originate from Loop directly?

Reply:
That's what I mean by "non-device alerts".  They don't use the new Alert Facility yet.